### ISSUE TITLE
Error while trying to build f18 using clang+libc++ on Ubuntu 19.04

### ISSUE THREAD

**Issue by ChinouneMehdi** [flang-compiler/f18#757]
While trying to build f18 on Ubuntu 19.04 using clang-8+libc++
```cmake .. -DCMAKE_CXX_COMPILER=clang++ -DBUILD_WITH_CLANG_LIBRARIES=/usr/lib/llvm-8```
I got this error message:
```
[ 64%] Linking CXX executable quick-sanity-test
clang: error: argument unused during compilation: '-nostdinc++' [-Werror,-Wunused-command-line-argument]
make[2]: *** [test/decimal/CMakeFiles/quick-sanity-test.dir/build.make:85: test/decimal/quick-sanity-test] Error 1
make[1]: *** [CMakeFiles/Makefile2:535: test/decimal/CMakeFiles/quick-sanity-test.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

**Comment by jeanPerier** [comment]
This comes from f18 build using `-Werror` for future proofing purposes. The build system white-lists a few warnings that are ok (such as the one causing your error).
When `-DBUILD_WITH_CLANG` is set, `-Wno-unused-command-line-argument` is added to `CMAKE_CXX_FLAGS`. In your case, you did not use `-DBUILD_WITH_CLANG` and instead directly defined clang using `CMAKE_CXX_COMPILER` so this warning was not disabled.

Maybe we should move this warning disabling to a part that uses a more generic detection of clang being the CXX compiler. It needs some testing.

In the meantime, you should be able to build f18, either by using `-DBUILD_WITH_CLANG` (see [Readme](https://github.com/flang-compiler/f18/blob/master/README.md)) or by manually adding -`DCMAKE_CXX_FLAGS="-Wno-unused-command-line-argument"` to your cmake command.

**Comment by ChinouneMehdi** [comment]
I found a workaround :
``` cmake .. -DBUILD_WITH_CLANG=/usr -DBUILD_WITH_CLANG_LIBRARIES=/usr/lib/llvm-8```
